### PR TITLE
Add logic to prevent shake detection when on background.

### PR DESCRIPTION
**Pull request by vic-gonzi**
With these changes, shake detector can be still started from Application class, and disabled when going to background.

As required in the issue description (https://github.com/pedrovgs/Lynx/issues/14), this feature is configurable, and permanent shake can remain on place. In this case, to prevent launching more than one activity instance, shake detector will be disabled immediately after launching the first one.

### CODE REVIEW

File: lynx/src/main/java/com/github/pedrovgs/lynx/LynxShakeDetector.java [hunk 1]
```diff
@@ -32,9 +32,11 @@
   private final Context context;
 
   private static boolean isEnabled = true;
+  private static boolean keepAlwaysActive;
 
-  public LynxShakeDetector(Context context) {
+  public LynxShakeDetector(Context context, boolean keepAlwaysActive) {
```
Comment (pedrovgs) [inline lynx/src/main/java/com/github/pedrovgs/lynx/LynxShakeDetector.java]: As this class is public adding a new param to the constructor without keeping the previous one means that we are introducing a breaking change. To improve the library adoption for new versions we should keep the old constructor calling this one with the default expected value.

File: lynx/src/main/java/com/github/pedrovgs/lynx/LynxShakeDetector.java [hunk 2]
```diff
@@ -56,17 +61,25 @@ public void init() {
...
-  static void enable() {
+  public static void enable() {
     isEnabled = true;
   }
 
   /**
    * Disables shake detector to open LynxActivity on shake.
    */
-  static void disable() {
+  public static void disable() {
```
Comment (pedrovgs) [inline lynx/src/main/java/com/github/pedrovgs/lynx/LynxShakeDetector.java]: These two public methods here are a smell telling us that this solution is not the best approach.

File: lynx/src/main/java/com/github/pedrovgs/lynx/LynxShakeDetector.java [hunk 3]
```diff
@@ -46,6 +48,9 @@ public void init() {
       @Override public void hearShake() {
         if (isEnabled) {
           openLynxActivity();
+          if (keepAlwaysActive) {
+            disable();
+          }
```
Comment (pedrovgs) [inline lynx/src/main/java/com/github/pedrovgs/lynx/LynxShakeDetector.java]: Instead of enable and disable the component why don't we just check if the application is in background before to open ``LynxActivity``?

File: lynx/src/main/java/com/github/pedrovgs/lynx/LynxShakeDetector.java [hunk 4]
```diff
@@ -32,9 +32,11 @@
   private final Context context;
 
   private static boolean isEnabled = true;
+  private static boolean keepAlwaysActive;
 
-  public LynxShakeDetector(Context context) {
+  public LynxShakeDetector(Context context, boolean keepAlwaysActive) {
     this.context = context;
+    this.keepAlwaysActive = keepAlwaysActive;
```
Comment (pedrovgs) [inline lynx/src/main/java/com/github/pedrovgs/lynx/LynxShakeDetector.java]: The name of this param should be something like: ``listenShakesOnBackground``

File: sample/src/main/java/com/github/pedrovgs/sample/MainActivity.java [hunk 5]
```diff
@@ -74,6 +75,20 @@ private void openLynxActivity() {
...
+    super.onPause();
+  }
+
+  @Override protected void onResume() {
+    super.onResume();
+    if (!LynxShakeDetector.isKeepAlwaysActive()) {
+      LynxShakeDetector.enable();
+    }
+  }
+
```
Comment (pedrovgs) [inline sample/src/main/java/com/github/pedrovgs/sample/MainActivity.java]: If you read my previous comments you'll see that we can revert these changes.

File: sample/src/main/java/com/github/pedrovgs/sample/LynxApplication.java [hunk 6]
```diff
@@ -29,7 +29,7 @@
 
   @Override public void onCreate() {
     super.onCreate();
-    LynxShakeDetector lynxShakeDetector = new LynxShakeDetector(this);
+    LynxShakeDetector lynxShakeDetector = new LynxShakeDetector(this, false);
```
Comment (pedrovgs) [inline sample/src/main/java/com/github/pedrovgs/sample/LynxApplication.java]: Once the default constructor be implemented remember to rever this change.

File: lynx/src/main/java/com/github/pedrovgs/lynx/LynxShakeDetector.java [hunk 7]
```diff
@@ -32,9 +32,11 @@
   private final Context context;
 
   private static boolean isEnabled = true;
+  private static boolean keepAlwaysActive;
```
Comment (pedrovgs) [inline lynx/src/main/java/com/github/pedrovgs/lynx/LynxShakeDetector.java]: This should not be static.

File: lynx/src/main/java/com/github/pedrovgs/lynx/LynxShakeDetector.java [hunk 8]
```diff
@@ -56,17 +61,25 @@ public void init() {
...
     isEnabled = false;
   }
 
+  /**
+   * Gets keep always active setting.
+   * @return true if shake detector should be always active, false otherwise.
+   */
+  public static boolean isKeepAlwaysActive() {
+    return keepAlwaysActive;
+  }
```
Comment (pedrovgs) [inline lynx/src/main/java/com/github/pedrovgs/lynx/LynxShakeDetector.java]: We can remove this method. The attribute should be used just from this class.

**Comment by pedrovgs** [comment]
Hi @vic-gonzi thanks for all your time and effort. All these pull requests (even the one I've already merged) are awesome. If you could update your branch with master fixing the merge conflicts and review my comments that'd be great!!!!!

File: lynx/src/main/java/com/github/pedrovgs/lynx/LynxShakeDetector.java (hunk 3 above)
Comment (vic-gonzi) [inline lynx/src/main/java/com/github/pedrovgs/lynx/LynxShakeDetector.java]: This change is not exactly for that. Imagine you start the app with the 'always active' flag enabled. Then you will want a LynxActivity launched if you shake the device even in background, but just one. That's why I've added this check.

File: sample/src/main/java/com/github/pedrovgs/sample/MainActivity.java (hunk 5 above)
Comment (pedrovgs) [inline sample/src/main/java/com/github/pedrovgs/sample/MainActivity.java]: Ok, we can update the min version supported to API 14 :)

**Comment by vibinr** [comment]
Waiting for this to be merged! 😄
